### DISCUSSION TITLE
What are the pros/cons of using the Streaming API?

### DISCUSSION THREAD

**Discussion by timvdhoorn** [teslamate-org/teslamate#3970] [Q&A]
As the title says. What are the pros/cons of using the Streaming API?
And what if I disable the Streaming API?

**Comment by jlestel** [comment]
Streaming is ideal for live data but only contains so many fields (12 before and 20 max with new Telemetry).

If you disable the streaming API, you will use the owner (or new fleet) API a lot more and risk being quickly banned.

**Reply by timvdhoorn** [reply]
So you recommend leaving streaming API enabled?

The main reason that I'm asking is when I disable the streaming API, in the car the location icon stays on all the time (i prefer it that way). 
But when I enable Streaming API this icon is constantly appearing/disappearing which makes the clock bounce which is annoying.
 I just implemented the MyTeslaMate API and haven't tested it yet with this way > if I disable streaming API it will only use the MyTeslaMate API, correct?

**Reply by jlestel** [reply]
> So you recommend leaving streaming API enabled?

Yep because it can give more data than Fleet API.

> if I disable streaming API it will only use the MyTeslaMate API, correct?

Yep but you will quickly reach the Fleet API quotas. That's why we're working on [#3940](https://github.com/teslamate-org/teslamate/pull/3940), to be able to increase API interval and keep streaming with will give more complete data in the end. By only using the Fleet API you risk having data for 1 hour before getting banned for the day.

**Comment by jlestel** [comment]
> But when I enable Streaming API this icon is constantly appearing/disappearing which makes the clock bounce which is annoying.

Indeed it is annoying that clock bounce every 10 seconds...

@cwanja @JakobLichterfeld @DrMichael : do you know if this is a "normal" behavior that all Teslamate (with streaming) users have?

**Reply by JakobLichterfeld** [reply]
> do you know if this is a "normal" behavior that all Teslamate (with streaming) users have?

All users using any sort of third party accessing location does have this behavior, as Tesla indicates the location access with it since 2023.xx.xx. Clock is not bouncing at all

**Reply by cwanja** [reply]
> > But when I enable Streaming API this icon is constantly appearing/disappearing which makes the clock bounce which is annoying.
> 
> Indeed it is annoying that clock bounce every 10 seconds...
> 
> @cwanja @JakobLichterfeld @DrMichael : do you know if this is a "normal" behavior that all Teslamate (with streaming) users have?

Yes, it is normal behavior as @JakobLichterfeld mentioned. 

I **do** notice in my Model S that the "menu bar notification" shows and then hides and repeats, causing the clock and weather icons to "flash". This is a root cause of TeslaMate, but Tesla's implementation of their location services notification icon.

**Reply by DrMichael** [reply]
I don't think I saw that on my M3... Cannot chekc at the moment.

**Reply by cwanja** [reply]
> I don't think I saw that on my M3... Cannot chekc at the moment.

Can try and get a video, do not recall if it happens when I am parked. Should. But honestly, I never pay attention to it. I wish Tesla gave me an option to turn it off.

**Reply by jlestel** [reply]
Thanks for your feedback! I was just curious to understand why the clock does not bounce for everyone.

I am trying to get a video to be able to compare with vehicles for which it does not "flash".

**Reply by JakobLichterfeld** [reply]
I am not sure what the purpose will be. You can either use TeslaMate or not. Live with the location access icon or not. Use different API and miss a lot of data, choice is yours.

**Reply by cwanja** [reply]
Here's the 'clock' bouncing around. Again, this is purely Tesla's issue. This icon would show up because of any third-party service polling the location. Happy to grab a longer video,  but just imagine it happening every 10/15 seconds. 

2021 (refreshed) Model S

https://github.com/user-attachments/assets/a8935ce5-0526-4144-ae31-02efd57a8d71

**Reply by cwanja** [reply]
@timvdhoorn is the video above what you are referring to?

**Reply by JakobLichterfeld** [reply]
For the sake of clarity, the clock will not bounce, but will be in horizontal motion. A bit different on M3 than in the above video but similar.

And yes this is expected behavior.

**Reply by timvdhoorn** [reply]
For me, the clock and the location icon is on the left side of the screen. Because of this it is a lot more anoying than the video, but I got used to it..

**Reply by jlestel** [reply]
> on the left side of the screen. Because of this it is a lot more anoying

Thanks for your feedback, indeed the side must explain why everyone does not have the same feeling about these icons.

**Reply by brianmay** [reply]
Why do people call this icon a clock? To me it looks nothing like a clock, it is a mobile phone with a nav icon.

Wondering where the 10/15 seconds comes from. Based on the comments it doesn't appear to be connected with the polling API. If this is to try to warn the occupants of the car that they are being tracked it is badly thought out in that case, a stalker would just use the polling API.

But maybe connected to something else, e.g. when I last tested this streaming API it had a habit of randomly disconnecting for no apparent reason, requiring a re-connection. Maybe this is happening every 10 seconds?

Alternatively, it could have been deliberately made annoying so the occupants will somebody is tracking them.

**Reply by cwanja** [reply]
> Why do people call this icon a clock? To me it looks nothing like a clock, it is a mobile phone with a nav icon.

The _clock_ is 'bouncing' because of the mobile phone / location services icon. 

> Alternatively, it could have been deliberately made annoying so the occupants will somebody is tracking them.

Would not put it past Tesla. But also annoying while you are driving 70MPH down the freeway.

**Reply by brianmay** [reply]
To me it looks like the problem is that the icon is added/removed on a frequent basis. Which is bad and probably easy for them to fix. However it is very likely nobody at Tesla uses anything like Teslamate, so they probably haven't noticed the problem. And we aren't in a good position to raise the issue either - it might draw attention to our use of the unofficial streaming API.

Does the problem occur when monitoring the location with the official app (and teslamate off)? If so, might have better luck complaining about that as the root cause.

**Comment by timvdhoorn** [comment]
For the ones that are wondering how it looks like on my side.

https://github.com/user-attachments/assets/d24411b7-ff6b-4e4a-bedd-3773b23a4346

**Comment by brianmay** [comment]
For reference, on 2016 Model S with updated MCU situation isn't as bad. Nav Icon turns on/off - which perhaps is slightly irritating, but it doesn't affect any of the other widgets.
